p2p/pex_reactor.go

```diff
@@ -276,8 +324,17 @@ func (r *PEXReactor) ensurePeers() {
 	for _, item := range toDial {
 		go func(picked *NetAddress) {
 			_, err := r.Switch.DialPeerWithAddress(picked, false)
```

Comment:
this returns `peer, err`. why not just use it instead of `peer := r.addrToSrcPeer(picked)`?

Reply:
addrToSrcPeer gets the source peer that told us about the peer we're trying to dial.

Reply:
exactly